src/features/SignAndVerifyMessage/signMessage.slice.ts

```diff
@@ -0,0 +1,99 @@
+import { getAddress } from '@ethersproject/address';
+import { createAction, createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { all, call, put, takeLatest } from 'redux-saga/effects';
+
+import { IFullWallet } from '@services/WalletService';
+import type { AppState } from '@store';
+import { ISignedMessage, WalletId } from '@types';
+import { addHexPrefix } from '@utils';
+
+interface State {
+  status: 'SIGN_SUCCESS' | 'SIGN_REQUEST' | 'SIGN_FAILURE' | 'INIT';
```

Comment:
Can you move this to an enum type?

Reply:
This is an intentional preference for Union type of string literals over enums.
https://blog.bam.tech/developer-news/should-you-use-enums-or-union-types-in-typescript

Since each status is exclusif of each other and we don't need to iterate over them, I think it's preferable over enums (especially if use integers as did the previous code).